error.go

```diff
@@ -71,30 +71,34 @@ type position struct {
 	byteNo int
 }
 
-func positionFromHeaders(headers map[uint16]string) (position, bool) {
+func positionFromHeaders(headers map[uint16]string) (position, bool, error) {
 	lineNoRaw, ok := headers[lineStart]
 	if !ok {
-		return position{}, false
+		return position{}, false, nil
 	}
 
 	byteNoRaw, ok := headers[positionStart]
 	if !ok {
-		return position{}, false
+		return position{}, false, nil
 	}
 
 	lineNo, err := strconv.Atoi(lineNoRaw)
 	if err != nil {
-		return position{}, false
+		return position{}, false, &binaryProtocolError{
+			msg: fmt.Sprintf("decode lineNo: %q: %s", lineNoRaw, err.Error()),
```

Comment:
minor and very nit-picky on my part, can this wrap the error?
```suggestion
			err: fmt.Errorf("decode lineNo: %q: %w", lineNoRaw, err),
```

Reply:
Sure :slightly_smiling_face: